Sources/NIOPosix/Bootstrap+vsock.swift

```diff
@@ -0,0 +1,48 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+#if canImport(Darwin) || os(Linux)
```

Comment:
unnecessary `#if`. Also, there is public API `#if`'d which shouldn't be done

Reply:
thanks so much! Yes, unavailable API is terrible for users because they'd need to replicate the `#if`s. If it doesn't work and throws an error that's much better.

You know if you tell Vapor to listen on a vsock on some unsupported platform and you get "unsupportedOperation" that's cool. If Vapor doesn't build or everybody has to perfectly copy the `#if`s into higher and higher level libraries/apps then that sucks.